sim-administration/simmanager/src/integration-test/kotlin/org/ostelco/simcards/admin/SimAdministrationTest.kt

```diff
@@ -303,64 +308,191 @@ class SimAdministrationTest {
-        val hlrs = simDao.getHssEntries()
-        assertThat(hlrs.isRight()).isTrue()
+        val pollingTask : PollOutstandingProfilesTask
 
-        var hssId: Long = 0
-        hlrs.map {
-            hssId = it[0].id
+        var hssId: Long = -1L
+
+        init {
+            testClass.loadSimData()
+
+            simDao = SIM_MANAGER_RULE.getApplication<SimAdministrationApplication>()
+                    .getDAO()
+
+            profileVendors = SIM_MANAGER_RULE.configuration.profileVendors
+            hssConfigs = SIM_MANAGER_RULE.configuration.hssVendors
+            httpClient = HttpClientBuilder(SIM_MANAGER_RULE.environment)
+                    .build("taskInvocationFixtureClient-${UUID.randomUUID()}")
+
+            val hlrs = simDao.getHssEntries()
+            assertThat(hlrs.isRight()).isTrue()
+
+            hlrs.map {
+                hssId = it[0].id
+            }
+
+            dispatcher = DirectHssDispatcher(
+                    hssConfigs = hssConfigs,
+                    httpClient = httpClient,
+                    healthCheckRegistrar = object : HealthCheckRegistrar {
+                        override fun registerHealthCheck(name: String, healthCheck: HealthCheck) {
+                            SIM_MANAGER_RULE.environment.healthChecks().register(name, healthCheck)
+                        }
+                    })
+            hssAdapterCache = SimManagerToHssDispatcherAdapter(
+                    dispatcher = dispatcher,
+                    simInventoryDAO = simDao)
+            preStats = SimProfileKeyStatistics(
+                    noOfEntries = 0L,
+                    noOfEntriesAvailableForImmediateUse = 0L,
+                    noOfReleasedEntries = 0L,
+                    noOfUnallocatedEntries = 0L,
+                    noOfReservedEntries = 0L)
+            pvaf = ProfileVendorAdapterFactory(
+                    simInventoryDAO = simDao,
+                    httpClient = httpClient,
+                    profileVendors = ConfigRegistry.config.profileVendors)
+            preallocationTask = PreallocateProfilesTask(
+                    maxNoOfProfileToAllocate = maxNoOfProfilesToAllocate,
+                    simInventoryDAO = simDao,
+                    hssAdapterProxy = hssAdapterCache,
+                    pvaf = pvaf)
+            simApi = SimInventoryApi(httpClient, SIM_MANAGER_RULE.configuration, simDao)
+
+            pollingTask = PollOutstandingProfilesTask(simInventoryDAO = simDao, pvaf = pvaf)
+        }
+
+
+
+        private fun executeTask(task: Task): String {
+            val out = StringWriter();
+            val writer = PrintWriter(out);
+            val parameters = ImmutableMultimap.builder<String, String>().build()
+            task.execute(parameters, writer)
+            return out.toString()
+        }
+
+        /**
+         * Execute the preallocation task, returning the report it makes as a string.
+         */
+        fun executePreallocateSimProfilesTask(): String {
```

Comment:
Such one liners can be written as
    fun executePreallocateSimProfilesTask(): String = executeTask(preallocationTask)
Improves readability.